Bottom: 1bdf2f18f526077f4c3f5fa8edbe0ddee085200b
Top:    e6ad7e2c764004b56417a83989ea1d9844c53e93
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-09 12:27:20 +0000

Refresh of attempt-to-fix-slowness

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9e8fbbb..149f453 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1854,7 +1854,7 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
 //                printf ("5: ");
-		PutContour (e, tmp, contours, holes, NULL, NULL, NULL); /* b */
+		PutContour (e, tmp, contours, holes, b, NULL, NULL); /* b */
 		break;
 	      case PBO_UNITE:
 		break;		/* nothing to do - already included */
@@ -1871,7 +1871,7 @@ M_B_AREA_Collect (jmp_buf * e, POLYAREA * bfst, POLYAREA ** contours,
 		tmp->next = NULL;
 		tmp->Flags.status = UNKNWN;
 //                printf ("6: ");
-		PutContour (e, tmp, contours, holes, NULL, NULL, NULL); /* b */
+		PutContour (e, tmp, contours, holes, b, NULL, NULL); /* b */
 		break;
 	      case PBO_ISECT:
 	      case PBO_SUB:
